test/entity/entity_test.go

```diff
@@ -0,0 +1,219 @@
+		agentConfigPath string
+		podName         string
+		useEC2Tag       bool
+		expectedEntity  expectedEntity
+	}{
+		"Entity/K8sWorkloadServiceNameSource": {
+			agentConfigPath: filepath.Join("resources", "compass_default_log.json"),
+			podName:         "log-generator",
+			expectedEntity: expectedEntity{
+				entityType:        "Service",
+				name:              "log-generator",
+				environment:       "eks:" + env.EKSClusterName + "/" + "default",
+				platformType:      "AWS::EKS",
+				k8sWorkload:       "log-generator",
+				k8sNode:           *instancePrivateDNS,
+				k8sNamespace:      "default",
+				eksCluster:        env.EKSClusterName,
+				instanceId:        env.InstanceId,
+				serviceNameSource: "K8sWorkload",
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var podApplicationLogStream string
+
+			appLogGroup := fmt.Sprintf("/aws/containerinsights/%s/%s", env.EKSClusterName, "application")
+			logStreamNames := awsservice.GetLogStreamNames(appLogGroup)
+			assert.NotZero(t, len(logStreamNames))
+			for _, streamName := range logStreamNames {
+				if strings.Contains(streamName, testCase.podName) {
+					podApplicationLogStream = streamName
+					log.Printf("Found log stream %s that matches pattern %s", streamName, testCase.podName)
+				}
+			}
+			assert.NotEmpty(t, podApplicationLogStream)
+			// check CWL to ensure we got the expected entities in the log group
+			queryString := fmt.Sprintf("fields @message, @entity.KeyAttributes.Type, @entity.KeyAttributes.Name, @entity.KeyAttributes.Environment, @entity.Attributes.PlatformType, @entity.Attributes.EKS.Cluster, @entity.Attributes.K8s.Node, @entity.Attributes.K8s.Namespace, @entity.Attributes.K8s.Workload, @entity.Attributes.AWS.ServiceNameSource, @entity.Attributes.EC2.InstanceId | filter @logStream == \"%s\"", podApplicationLogStream)
+			ValidateEntity(t, appLogGroup, podApplicationLogStream, &end, queryString, testCase.expectedEntity, string(env.ComputeType))
+		})
+	}
+}
+
+// ValidateEntity queries a given LogGroup/LogStream combination given the start and end times, and executes an
+// log query for entity attributes to ensure the entity values are correct
+func ValidateEntity(t *testing.T, logGroup, logStream string, end *time.Time, queryString string, expectedEntity expectedEntity, entityPlatformType string) {
+	var requiredEntityFields map[string]bool
+
+	log.Printf("Checking log group/stream: %s/%s", logGroup, logStream)
+
+	if !awsservice.IsLogGroupExists(logGroup) {
+		t.Fatalf("application log group used for entity validation doesn't exsit: %s", logGroup)
+	}
+	begin := end.Add(-sleepForFlush * 2)
+	log.Printf("Start time is " + begin.String() + " and end time is " + end.String())
+
+	result, err := awsservice.GetLogQueryResults(logGroup, begin.Unix(), end.Unix(), queryString)
+	assert.NoError(t, err)
+	if !assert.NotZero(t, len(result)) {
+		return
+	}
+	if entityPlatformType == "EC2" {
```

Comment:
Feel like we should have a defined constant for required KeyAttributes and Attributes for each of the platforms. 

Something like

```
ec2RequiredKeyAttributes := map[string]struct{}{} {
    "Service": {},
    "Environment":{},
    ...
}
```

And have some validator in a util func. Having a map[string]struct{}{} is just easier to do the checks...not sure if we have a List abstraction for this.

Reply:
Going to create a validator struct that performs different logics based on platforms.